### PR TITLE
stop previous apolloServer before creating a new one

### PR DESCRIPTION
**Pull request by arash16**
This causes socket listeners to close gracefully.
Fixes issues #51 and #74

### CODE REVIEW

File: src/service.js [hunk 1]
```diff
@@ -612,6 +612,10 @@ module.exports = function (mixinOptions) {
 				if (!this.shouldUpdateGraphqlSchema && this.graphqlHandler) {
 					return;
 				}
+				
+				if (this.apolloServer) {
+					this.apolloServer.stop();
```
Comment (shawnmcknight) [inline src/service.js:617]: I'm concerned here as Apollo's docs specify this as an `async` method but there is no promise handling on this call.

https://www.apollographql.com/docs/apollo-server/api/apollo-server/#stop

Comment (Hugome) [reply src/service.js:617]: Shared concerns.
The stop function will gracefully close the subscription server and so the socket and if you're unlucky the new sever will try listen before the old one is fully closed
(https://github.com/apollographql/apollo-server/blob/b7a91df76acef748488eedcfe998917173cff142/packages/apollo-server-core/src/ApolloServer.ts#L594)

Comment (arash16) [reply src/service.js]: fixed

**Comment by shawnmcknight** [comment]
@arash16 CI is currently failing.  Tests need to be updated to handle your changes.

**Comment by shawnmcknight** [comment]
I've updated the tests so they are no longer failing.
@Hugome or @icebob Would you mind giving this a review?  Everything looks good to me now.

**Review by Hugome — APPROVED** [review]
LGTM

**Review by icebob — APPROVED** [review]
I can't try it now, but the code looks good to me, too.

**Pull request merged**